docs-src/how-to-create-layoutbuilder.md

```diff
@@ -14,10 +14,91 @@ kernelspec:
 
-If you need it soon, create an issue saying so and I'll make it a higher priority.
+ak::LayoutBuilder can help you to create these "low-level" views that are described by a ak::Form. Once the builder is initialized, it can only build a specific view determined by the layout form.
 
-[![](img/github-issues-documentation.png)](https://github.com/scikit-hep/awkward-1.0/issues/new?assignees=&labels=docs&template=documentation.md&title=)
+The biggest difference between a #ak::LayoutBuilder and an [ak.ArrayBuilder](https://awkward-array.readthedocs.io/en/latest/_auto/ak.ArrayBuilder.html) is that the data types that you can append to the #ak::LayoutBuilder are restricted by its #ak::Form, while you can append any data types to an #ak::ArrayBuilder. The latter flexibility comes with performance limitations.
 
-The text of your issue doesn't have to be much more than a link to this page, so I can be sure which page you're referring to. If you add details about how and why you need it, however, I may be able to tailor the text to help you more.
++++
+
+Appending
+---------
+
+  ```{code-cell}
+  import awkward as ak
+  ```
+To create an #ak::LayoutBuilder an #ak::Form in JSON format is needed to initialize it. This #ak::Form determines which commands and which data types are accepted by the builder.
+
+Here is an example of a form describing an #ak.layout.UnionArray8_64 of a [union-type](https://awkward-array.readthedocs.io/en/latest/ak.types.UnionType.html):
+
+```{code-cell}
+form = """
+{
+  "class": "UnionArray8_64",
+  "tags": "i8",
+  "index": "i64",
+  "contents": [
+      "float64",
+      "bool",
+      "int64"
+  ],
+  "form_key": "node0"
+}
+  """
+```
+
+When a builder is created, it will accept only "float64", "bool", and "int64" data. The builder data methods to append these types have similar to the types names: `float64`, `boolean`, and `int64`.
+
+```{code-cell}
+builder = ak.layout.LayoutBuilder(form)
+```
+
+Each of the union array contents has a tag associated with it. The tags are contiguous integers starting with `0` for the first content. You have to issue a `tag` command prior to each data method. The order of filling the contents can be arbitrary - the `tag` command uniquely identifies the content to which the data is appended:
+
+```{code-cell}
+builder.tag(0)
+builder.float64(1.1)
+builder.tag(1)
+builder.boolean(False)
+builder.tag(2)
+builder.int64(11)
+builder.tag(0)
+builder.float64(2.2)
+builder.tag(1)
+builder.boolean(False)
+builder.tag(0)
+builder.float64(2.2)
+builder.tag(0)
+builder.float64(3.3)
+builder.tag(1)
+builder.boolean(True)
+builder.tag(0)
+builder.float64(4.4)
+builder.tag(1)
+builder.boolean(False)
+builder.tag(1)
+builder.boolean(True)
+builder.tag(0)
+builder.float64(-2.2)
+```
+
+Snapshot
+--------
+
+To turn a #ak::LauoutBuilder into a layout, call `snapshot`. This is an inexpensive operation (may be done multiple times; the builder is unaffacted).
```

Comment:
```suggestion
To turn a #ak::LayoutBuilder into a layout, call `snapshot`. This is an inexpensive operation (may be done multiple times; the builder is unaffected).
```

Reply:
I took a look over this out of curiosity - let me know if you'd rather wait for review before any suggestions. I can't do much besides code review at the moment so I'm keen to help!

Reply:
thanks, I'll pin you when I'm finished!

Reply:
@agoose77 - Can I take you up on your kind offer of code review? Thanks!